app/web/features/search/SearchBox.tsx

```diff
@@ -40,186 +29,104 @@ const useStyles = makeStyles((theme) => ({
-      }, theme.transitions.duration.standard);
-    }
-    //necessary because we don't want to cache every search for each filter
-    //but we do want react-query to handle pagination
-    queryClient.removeQueries(searchQueryKey());
-    searchFilters.apply();
-  };
 
-  const handleKeywordsChange = (
-    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement> | ""
-  ) => {
-    searchFilters.remove("location");
-    searchFilters.remove("lat");
-    searchFilters.remove("lng");
-    searchFilters.remove("bbox");
-
-    setValue("location", "");
-
-    if (event === "") {
-      searchFilters.remove("query");
-    } else {
-      if (event.target.value === "") {
-        searchFilters.remove("query");
-      } else {
-        searchFilters.change("query", event.target.value);
-      }
-    }
-    //necessary because we don't want to cache every search for each filter
-    //but we do want react-query to handle pagination
-    queryClient.removeQueries(searchQueryKey());
-    searchFilters.apply();
-  };
-  const handleKeywordsChangeDebounced = debounce(handleKeywordsChange, 500);
-
-  //in case the filters were changed in the dialog, update here
-  useEffect(() => {
-    setValue("location", searchFilters.active.location ?? "");
-    setValue("query", searchFilters.active.query ?? "");
-  }, [setValue, searchFilters.active.location, searchFilters.active.query]);
-
-  const filterDialog = (
-    <FilterDialog
-      updateMapBoundingBox={updateMapBoundingBox}
-      isOpen={isFiltersOpen}
-      onClose={() => setIsFiltersOpen(false)}
-      searchFilters={searchFilters}
-    />
-  );
-
-  if (isSmDown) {
-    return (
-      <>
-        <Button
-          onClick={() => setIsFiltersOpen(true)}
-          className={className}
-          variant="contained"
-          size="medium"
-        >
-          {t("search:filter_dialog.mobile_title")}
-        </Button>
-
-        {filterDialog}
-      </>
-    );
-  }
+  const { control, setValue, errors, register, handleSubmit } = useForm({
+    mode: "onChange",
+  });
 
   return (
     <>
       {searchType === "location" ? (
         <LocationAutocomplete
           control={control}
           name="location"
-          defaultValue={
-            searchFilters.active.location
-              ? {
-                  name: searchFilters.active.location,
-                  simplifiedName: searchFilters.active.location,
-                  location: new LngLat(
-                    searchFilters.active.lng ?? 0,
-                    searchFilters.active.lat ?? 0
-                  ),
-                  bbox: searchFilters.active.bbox ?? [0, 0, 0, 0],
-                }
-              : ""
-          }
+          defaultValue={locationResult.name}
           label={t("search:form.location_field_label")}
-          onChange={handleNewLocation}
+          onChange={(event) => {
+            if (event) {
+              setLocationResult(event);
+            }
+          }}
```

Comment:
Just a nit - but using an inline function will create a new instance of the function every time it runs. Putting it into `handleOnChange` would technically prevent some re-renders here, I believe. But not super important unless it's processing huge amounts of data.

Reply:
ok, maybe it's important! didn't know that creating a new function could trigger a new re-render.... (I kind of hate this re-render thing, seems kind of complex) will update it!!

Reply:
(I don't think it's a nit since we pretend to add way more features to map, so any performance  improvement it's welcome!)

Reply:
updated ✅